godot_project/project_workspace/structs/dna_structure.gd

```diff
@@ -48,16 +48,23 @@ func grab_curve(out_path3d: Path3D) -> void:
 	out_path3d.curve = _curve
 
 
-func get_baked_path() -> PackedVector3Array:
-	if _baked_path.is_empty() or _signal_queue_path_changed:
-		_baked_path = _curve.get_baked_points()
+## Get a list of points representing the path
+## in_path_override can be provided to get an alternative path while editing the spline of this structure
+func get_baked_path(in_path_override: Curve3D = null) -> PackedVector3Array:
+	var curve: Curve3D = _curve if in_path_override == null else in_path_override
+	if _baked_path.is_empty() or _signal_queue_path_changed or in_path_override != null:
+		var baked_path: PackedVector3Array = curve.get_baked_points()
```

Comment:
/!\ get_baked_points() can return an empty array if the curve is empty (which would cause a crash on lines 59 - 60)

Reply:
i believe if all points where to be deleted, then the entire structure should be deleted, but having 1 point is likely an issue anyway